File: WebSite with CosmosDB/server/Basketball/Microservice/ViewModels/ParticipationVM.cs
using NBA.Infrastructure;
using NBA.Models;

namespace NBA.ViewModels
{
    public class ParticipationVM : BasketballViewModel
    {
        public string GameId { get; set; }

        public string PlayerId { get; set; }

        public List<GamePlay> Plays { get; set; } = [];

        public override (bool Success, string Message) Validate()
        {
            return (false, "this viewModel cannot be used for write operations");
        }

        internal static ParticipationVM FactorFrom(Participation model)
        {
            return new()
            {
                PlayerId = model.PlayerId,
                GameId = model.GameId,
                Plays = model.Plays
            };
        }
    }
}

File: Vue-Nuxt with Java-MongoDB/server/Basketball/Microservice/src/main/java/com/nba/microservice/viewmodels/ParticipationVM.java
package com.nba.microservice.viewmodels;

import com.nba.microservice.models.Participation;
import com.nba.microservice.models.ValueObjects.GamePlay;
import com.nba.microservice.infrastructure.BasketballViewModel;
import com.nba.microservice.infrastructure.ValidationResult;
import java.util.ArrayList;
import java.util.List;

public class ParticipationVM extends BasketballViewModel {

    private String participationId;
    private String gameId;
    private String playerId;
    private List<GamePlayVM> plays;

    public String getParticipationId() {
        return participationId;
    }

    public void setParticipationId(String participationId) {
        this.participationId = participationId;
    }

    public String getGameId() {
        return gameId;
    }

    public void setGameId(String gameId) {
        this.gameId = gameId;
    }

    public String getPlayerId() {
        return playerId;
    }

    public void setPlayerId(String playerId) {
        this.playerId = playerId;
    }

    public List<GamePlayVM> getPlays() {
        return plays;
    }

    public void setPlays(List<GamePlayVM> plays) {
        this.plays = plays;
    }

    @Override
    public ValidationResult validate() {
        return new ValidationResult(false, "this viewModel cannot be used for write operations");
    }

    public static ParticipationVM factoryFrom(Participation model) {
        ParticipationVM participationVM = new ParticipationVM();
        participationVM.setParticipationId(model.getId());
        participationVM.setPlayerId(model.getPlayerId());
        participationVM.setGameId(model.getGameId());

        List<GamePlayVM> gamePlayVMs = new ArrayList<>();
        for (GamePlay play : model.getPlays()) {
            gamePlayVMs.add(GamePlayVM.factoryFrom(play));
        }
        participationVM.setPlays(gamePlayVMs);

        return participationVM;
    }
}